internal/services/paloalto/schema/marketplace_details.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package schema
+
+import (
+	"github.com/hashicorp/go-azure-sdk/resource-manager/paloaltonetworks/2023-09-01/firewalls"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type MarketplaceDetails struct {
+	OfferId     string `tfschema:"offer_id"`
+	PublisherId string `tfschema:"publisher_id"`
+}
+
+func MarketplaceDetailsSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Optional: true,
```

Comment:
As "marketplace_details" is defined as optional, may I ask if the resource can be created successfully when it isn't specified? Seems Swagger indicates that it's a required property.